packages/app/components/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useMemo, useLayoutEffect } from "react";
+
+import { BottomSheetModalProvider } from "@gorhom/bottom-sheet";
+import { AnimatePresence } from "moti";
+
+import { useRouter } from "@showtime-xyz/universal.router";
+import { View } from "@showtime-xyz/universal.view";
+
+import { useUser } from "app/hooks/use-user";
+import { createParam } from "app/navigation/use-param";
+
+import { OnboardingStepContext } from "./onboarding-context";
+import { OnboardingStep, PageQuery } from "./onboarding-types";
+import { SelectPicture } from "./select-picture";
+import { SelectSocial } from "./select-social";
+import { SelectUsername } from "./select-username";
+
+const { useParam } = createParam<PageQuery>();
+
+export const Onboarding = () => {
+  const { user, isIncompletedProfile } = useUser();
+  const router = useRouter();
+
+  const [redirectUri] = useParam("redirectUri");
+
+  // redirect to home if user is not incompleted profile
+  // this should prevent user from going back to onboarding
+  // with the back button
```

Comment:
i think if we do router.replace post onboarding, we might not this special handling. wdyt?

Reply:
There are cases though which require router.pop, like the claim. Main reason was that I was calling an existing hook which used `push` to open the claim dialog. I could duplicate that hook though and make a replace version, or pass a param. WDYT? 

To be specific, the main reason is `useRedirectToClaimDrop`

Reply:
I added a second param `shouldUseReplace` to useRedirectToClaimDrop (default false) so I can reuse that hook and get rid of that special handling

Reply:
Even though I added this this logic https://github.com/showtime-xyz/showtime-frontend/pull/1925/commits/26755351e6ccaefe5a2a35e19042a7f4c54f298e#diff-20211b750f4af859e9d85d45e05e02fc483793752948b7b525cb18c2bf48abcbR46-R56 it still pushes on top instead of replacing it. Very odd

Reply:
I found another solution, so this is resolved now

Reply:
I thought about it once more and I decided to add this back, because even when we replace, users on web could still directly call /onboarding

Reply:
okay, let's keep it 👍